Shiny updates core patch

swissspidy opened the pull request:
Cheatin', uh?

I created this PR in order to easily create a diff file for the Shiny Updates feature plugin and to have Travis run our tests automatically.

Please ignore 🙂

More information:
-  [Core merge proposal](https://make.wordpress.org/core/2016/06/02/proposal-more-shiny-updates/)
- [Plugin on WordPress.org](https://wordpress.org/plugins/shiny-updates/)

swissspidy committed:
Begin with core patch

To do:

- Actually test this
- Port over all of the CSS

swissspidy committed:
Fix theme updates script handle

swissspidy committed:
Get rid of theme-updates.js

Incorporates the needed changes into theme.js

swissspidy committed:
Themes: Disable key events handler when filesystem credentials dialog is shown

Fixes obenland/shiny-updates#237

swissspidy committed:
Themes: Do not close details modal when closing the credentials dialog

See obenland/shiny-updates#237

swissspidy committed:
Remove triple space

swissspidy committed:
Remove errant space in `_get_list_table()` call

swissspidy committed:
Implement obenland/shiny-updates#239

swissspidy committed:
Enqueue updates script on import.php

swissspidy committed:
Enqueue updates script on import.php

swissspidy committed:
Run `wp_print_admin_notice_template()` where needed

swissspidy committed:
Run `wp_print_admin_notice_template()` where needed

swissspidy committed:
4.X.0 => 4.6.0

swissspidy committed:
4.X.0 => 4.6.0

swissspidy committed:
Add the rest of Shiny Updates CSS

Needs testing

swissspidy committed:
Add the rest of Shiny Updates CSS

Needs testing

swissspidy committed:
Add global declarations in QUnit tests

swissspidy committed:
Add global declarations in QUnit tests

swissspidy committed:
Fix typo in updates test file

swissspidy committed:
Fix typo in updates test file

swissspidy committed:
Require JS for "Update All" functionality

See obenland/shiny-updates#245

swissspidy committed:
Require JS for "Update All" functionality

See obenland/shiny-updates#245

swissspidy committed:
Allow re-installing dismissed updates

See obenland/shiny-updates#243

swissspidy committed:
Allow re-installing dismissed updates

See obenland/shiny-updates#243

swissspidy committed:
Use correct selector in `wp.updates.installThemeSuccess`

swissspidy committed:
Use correct selector in `wp.updates.installThemeSuccess`

swissspidy committed:
Hide theme install button after text change

swissspidy committed:
Hide theme install button after text change

swissspidy committed:
Re-render theme view on change

This change got missed when first starting with the patch. It ensures that the "Installed" notice is added to the theme after installing

swissspidy committed:
Re-render theme view on change

This change got missed when first starting with the patch. It ensures that the "Installed" notice is added to the theme after installing

swissspidy committed:
Change CSS order to remove spacing in themes overview

swissspidy committed:
Change CSS order to remove spacing in themes overview

obenland committed:
Disable failing screen reader test.

obenland committed:
Disable failing screen reader test.

obenland committed:
Update function docs for `wp.ajax`.

obenland committed:
Update function docs for `wp.ajax`.

obenland committed:
Update property/method docs as needed for ajax handlers.

obenland committed:
Update property/method docs as needed for ajax handlers.

obenland committed:
Rearrange ajax handlers to consolidate update handlers.

obenland committed:
Rearrange ajax handlers to consolidate update handlers.

obenland committed:
Proper handling of misc WP_Errors when updating themes.

obenland committed:
Proper handling of misc WP_Errors when updating themes.

obenland committed:
Merge latest changes from Shiny Updates master.

obenland committed:
Merge latest changes from Shiny Updates master.

obenland committed:
Group upgrade plugin callback with the other Shiny updates handlers.

obenland committed:
Group upgrade plugin callback with the other Shiny updates handlers.

obenland committed:
Merge obenland/shiny-updates#240.

obenland committed:
Merge obenland/shiny-updates#240.

obenland committed:
Add missing CSS selector.

Finishes merge of obenland/shiny-updates#251

obenland committed:
Add missing CSS selector.

Finishes merge of obenland/shiny-updates#251

obenland committed:
Add `bulk-action-form` for network themes.

Must have been lost in the shuffle at some point.
Fixes obenland/shiny-updates#269.

obenland committed:
Add `bulk-action-form` for network themes.

Must have been lost in the shuffle at some point.
Fixes obenland/shiny-updates#269.

obenland committed:
Maintain warning for network activated themes.

obenland committed:
Maintain warning for network activated themes.

obenland committed:
Disable "Update All" when all updates are completed

Fixes obenland/shiny-updates#266.

obenland committed:
Disable "Update All" when all updates are completed

Fixes obenland/shiny-updates#266.

obenland committed:
Don't break foreach loop.

Fixes a bug where active plugins would break the foreach loop.

obenland committed:
Don't break foreach loop.

Fixes a bug where active plugins would break the foreach loop.

obenland committed:
Account for network activations.

Fixes obenland/shiny-updates#254.

obenland committed:
Account for network activations.

Fixes obenland/shiny-updates#254.

obenland committed:
Repair ajaxified plugin searches.

obenland committed:
Repair ajaxified plugin searches.

obenland committed:
It takes more than button text to properly network activate a plugin.

Unfortunately we can’t just set the value of `network wide` to whatever
the result of `is_network_admin()` is, because the mere existence of
the query var is enough for WP to count it as network wide.

See
https://github.com/swissspidy/develop.wordpress/commit/f5c96e84ab33efb3d
49a0b3e52f4fb6aeadd82d7#commitcomment-17773179

obenland committed:
It takes more than button text to properly network activate a plugin.

Unfortunately we can’t just set the value of `network wide` to whatever
the result of `is_network_admin()` is, because the mere existence of
the query var is enough for WP to count it as network wide.

See
https://github.com/swissspidy/develop.wordpress/commit/f5c96e84ab33efb3d
49a0b3e52f4fb6aeadd82d7#commitcomment-17773179

swissspidy committed:
Merge branch 'master' into shiny-updates

obenland committed:
Make theme update notification work in Core

For some odd reason updating `hasUpdate` did not work in Shiny Updates
and we had to replace the model. That however doesn’t work in Core,
while updating the property does. Oh well.

obenland committed:
Make theme update notification work in Core

For some odd reason updating `hasUpdate` did not work in Shiny Updates
and we had to replace the model. That however doesn’t work in Core,
while updating the property does. Oh well.

obenland committed:
Use existing core handler for dismissible admin notices.

Fixes obenland/shiny-updates#210.

obenland committed:
Use existing core handler for dismissible admin notices.

Fixes obenland/shiny-updates#210.

obenland committed:
Merge remote-tracking branch 'aaronjorbin/master' into shiny-updates

obenland committed:
Merge remote-tracking branch 'origin/shiny-updates' into shiny-updates

obenland committed:
Update theme version in details modal.

H/t @ocean90.

See 35c019aa03ed670fd7bd2cb5ae0a0759cd6a67ad.

obenland committed:
Update theme version in details modal.

H/t @ocean90.

See 35c019aa03ed670fd7bd2cb5ae0a0759cd6a67ad.

obenland committed:
Make sure a delete theme is also removed from the original data and search.

obenland committed:
Make sure a delete theme is also removed from the original data and search.

swissspidy committed:
Add "Waiting..." text to core update button if it's actually waiting

swissspidy committed:
Add "Waiting..." text to core update button if it's actually waiting

obenland committed:
Use updated update notice on theme index.

obenland committed:
Use updated update notice on theme index.

obenland committed:
Add missing plugin install error event.

obenland committed:
Add missing plugin install error event.

obenland committed:
Disable plugin install link after fail install attempt.

obenland committed:
Disable plugin install link after fail install attempt.

obenland committed:
First pass at better CSS

Removes a lot of the CSS we had to overwrite in the plugin.

obenland committed:
First pass at better CSS

Removes a lot of the CSS we had to overwrite in the plugin.

swissspidy committed:
Move event listener out of the callback

swissspidy committed:
Move event listener out of the callback

obenland committed:
Finish up CSS fixes.

obenland committed:
Finish up CSS fixes.

swissspidy committed:
Update selector

swissspidy committed:
Update selector

obenland committed:
The queue contains more than just updates.

obenland committed:
The queue contains more than just updates.

obenland committed:
Namespace plugin success/error callbacks.

obenland committed:
Namespace plugin success/error callbacks.

obenland committed:
Remove unused global.

See 116d06a541005cca3c92806b681b5bdc4d0e3929.

obenland committed:
Remove unused global.

See 116d06a541005cca3c92806b681b5bdc4d0e3929.

obenland committed:
No waiting just updating.

obenland committed:
No waiting just updating.

obenland committed:
Complete PR

obenland committed:
Complete PR

swissspidy committed:
Merge pull request #1 from swissspidy/no-waiting-just-updating

No waiting just updating.

obenland committed:
Update property docs.

obenland committed:
Update property docs.

obenland committed:
This lock locks more than updates.

obenland committed:
This lock locks more than updates.

obenland committed:
Consolidate iframe install/update handlers.

obenland committed:
Consolidate iframe install/update handlers.

obenland committed:
Remove whitespace

obenland committed:
Remove whitespace

obenland committed:
Only display AYS message when deleting themes.

obenland committed:
Only display AYS message when deleting themes.

obenland committed:
Consolidate bulk action handler for plugins and themes.

obenland committed:
Consolidate bulk action handler for plugins and themes.

obenland committed:
Update inline docs for bulk action handler.

obenland committed:
Update inline docs for bulk action handler.

obenland committed:
Revert back to using an ID for better performance.

obenland committed:
Revert back to using an ID for better performance.

obenland committed:
Add missing semicolon.

obenland committed:
Add missing semicolon.

swissspidy committed:
Merge branch 'master' into shiny-updates

swissspidy committed:
If a core update is running, disable all other buttons

swissspidy committed:
If a core update is running, disable all other buttons

swissspidy committed:
Also disable all buttons if there's no other update left besides the current one

swissspidy committed:
Also disable all buttons if there's no other update left besides the current one

swissspidy committed:
Improve handling of "Update All"

swissspidy committed:
Improve handling of "Update All"

swissspidy committed:
Various selector improvements

swissspidy committed:
Various selector improvements

swissspidy committed:
s/updateLock/ajaxLocked

See 76eae560e3c53dac8fa7d214790d3d9b06dd58a2

swissspidy committed:
s/updateLock/ajaxLocked

See 76eae560e3c53dac8fa7d214790d3d9b06dd58a2

swissspidy committed:
DRY: Add `maybeRequestFilesystemCredentials` method

swissspidy committed:
DRY: Add `maybeRequestFilesystemCredentials` method

swissspidy committed:
Simplify length comparisons

swissspidy committed:
Simplify length comparisons

swissspidy committed:
`maybeRequestFilesystemCredentials` follow up

swissspidy committed:
`maybeRequestFilesystemCredentials` follow up

swissspidy committed:
Uncheck boxes on bulk action again

swissspidy committed:
Uncheck boxes on bulk action again

swissspidy committed:
Docs improvements

swissspidy committed:
Docs improvements

swissspidy committed:
DRY: Add `maybeHandleCredentialError` method

swissspidy committed:
DRY: Add `maybeHandleCredentialError` method

swissspidy committed:
Selector improvements

swissspidy committed:
Selector improvements

swissspidy committed:
s/updateQueue/queue

swissspidy committed:
s/updateQueue/queue

swissspidy committed:
Adjust color for "Update All" button icon

swissspidy committed:
Adjust color for "Update All" button icon

obenland committed:
Restore blank lines before inline comments and consolidate selectors.

obenland committed:
Restore blank lines before inline comments and consolidate selectors.

obenland committed:
Use the term `type` for the type of item we deal with.

obenland committed:
Use the term `type` for the type of item we deal with.

obenland committed:
Use the term `action` for the type of request to be performed.

obenland committed:
Use the term `action` for the type of request to be performed.

obenland committed:
Finalize `maybeHandleCredentialError` implementation.

obenland committed:
Finalize `maybeHandleCredentialError` implementation.

swissspidy commented on src/wp-admin/js/updates.js:
```diff
@@ -305,6 +305,10 @@
 				break;
+
+			default:
+				window.console.error( '"%s" is not white-listed to have its count decremented.', type );
```
Attenzione signore. `wp.updates.decrementCount` is called by `wp.updates.updateItemSuccess`, which handles any type (translation updates need to be decremented too!). You should be able to pass all valid types to that function, but for 'plugin' and 'theme' it does some more stuff.

ocean90 replied:
`window.console.error` should be wrapped by `if ( window.console && window.console.error ) { }`.

obenland committed:
Update plugin selectors to work after dynamic search.

obenland committed:
Update plugin selectors to work after dynamic search.

obenland committed:
Make decrementing counts on plugin/theme delete work again.

obenland committed:
Make decrementing counts on plugin/theme delete work again.

obenland committed:
Add Shiny Updates support to network themes.

H/t @afragen.

See
https://wordpress.slack.com/archives/feature-shinyupdates/p1465335873000
365

obenland committed:
Add Shiny Updates support to network themes.

H/t @afragen.

See
https://wordpress.slack.com/archives/feature-shinyupdates/p1465335873000
365

swissspidy committed:
Remove errant plus symbol

#blameobenland

swissspidy committed:
Remove errant plus symbol

#blameobenland

swissspidy committed:
Disable other buttons when re-installing core

swissspidy committed:
Disable other buttons when re-installing core

obenland committed:
Fix quit tests

Fix tests after 7a720596f803e415ae622d7a02ebe5db5b09e001.

obenland committed:
Fix quit tests

Fix tests after 7a720596f803e415ae622d7a02ebe5db5b09e001.

obenland committed:
Fix floating headline in network themes.

obenland committed:
Fix floating headline in network themes.

swissspidy committed:
Distinguish between bulk theme and plugin updates

swissspidy committed:
Distinguish between bulk theme and plugin updates

swissspidy committed:
Split `wp_print_admin_notice_template()` into two

swissspidy committed:
Split `wp_print_admin_notice_template()` into two

swissspidy committed:
Fix selector for `$allOtherUpdateButtons`

It should include the re-install button as well

swissspidy committed:
Fix selector for `$allOtherUpdateButtons`

It should include the re-install button as well

obenland committed:
Account for Update All when determining the type.

Introduced in  3bd8cd916f00c9f862432158a16a6e09aa78864b.

H/t @afragen

See
https://wordpress.slack.com/archives/feature-shinyupdates/p1465417602000
510

obenland committed:
Account for Update All when determining the type.

Introduced in  3bd8cd916f00c9f862432158a16a6e09aa78864b.

H/t @afragen

See
https://wordpress.slack.com/archives/feature-shinyupdates/p1465417602000
510

Commit:
ajax-actions.php improvements.

Commit:
ajax-actions.php improvements.

Commit:
No periods in file header summary.

Commit:
No periods in file header summary.

Commit:
WP_Updates_List_Table improvements.

Commit:
WP_Updates_List_Table improvements.

Commit:
deprecated.php improvements.

Commit:
deprecated.php improvements.

Commit:
update.php improvements.

Commit:
update.php improvements.

obenland committed:
Merge quit tests into existing structure.

Fixes obenland/shiny-updates#274

obenland committed:
Merge quit tests into existing structure.

Fixes obenland/shiny-updates#274

obenland committed:
Merge pull request #2 from DrewAPicture/shiny-updates

Core docs audit

obenland committed:
Allow themes to be activated from theme-install.

* Adds activate/live preview functionality to all installed themes in
theme-install, not only freshly installed themes.
* Networks enables instead of activates it in multisite.
* Doesn’t switch Customizer Preview links since there is no way to
customize a theme from network admin.

obenland committed:
Allow themes to be activated from theme-install.

* Adds activate/live preview functionality to all installed themes in
theme-install, not only freshly installed themes.
* Networks enables instead of activates it in multisite.
* Doesn’t switch Customizer Preview links since there is no way to
customize a theme from network admin.

obenland committed:
Merge branch 'shiny-updates' into feature/activate-installed-themes

# Conflicts:
#	src/wp-admin/includes/ajax-actions.php

obenland committed:
More specific naming activate message

Brings it in line with `activateTheme` and `activateImporter`.

obenland committed:
More specific naming activate message

Brings it in line with `activateTheme` and `activateImporter`.

obenland committed:
Don't remove focus after theme install.

Keeping the focus shows the added action of activating the theme.

obenland committed:
Don't remove focus after theme install.

Keeping the focus shows the added action of activating the theme.

obenland committed:
Don't unnecessarily replace install button.

obenland committed:
Don't unnecessarily replace install button.

swissspidy committed:
Merge pull request #3 from swissspidy/feature/activate-installed-themes

Allow themes to be activated from theme-install.

swissspidy committed:
Basic validation of AJAX responses

swissspidy committed:
Type column: s/Core/WordPress

swissspidy committed:
Type column: s/Core/WordPress

swissspidy committed:
Do not show compatibility note if plugin is fully compatible

If the plugin is 100% compatible with the current version of WordPress and the new WordPress version (if available), do not show a compatibility note.

swissspidy committed:
Do not show compatibility note if plugin is fully compatible

If the plugin is 100% compatible with the current version of WordPress and the new WordPress version (if available), do not show a compatibility note.

obenland committed:
Minor formatting update.

obenland committed:
Minor formatting update.

obenland committed:
Merge pull request #6 from swissspidy/core-type-column

Type column: s/Core/WordPress

swissspidy committed:
Merge master into branch

obenland committed:
Don't stomp $compat variable.

obenland committed:
Don't stomp $compat variable.

obenland committed:
Merge pull request #7 from swissspidy/maybe-hide-compat

Do not show compatibility note if plugin is fully compatible.

swissspidy committed:
Merge remote branch

swissspidy committed:
Revert c2ca7fc9d15686759b6c0ac1fd24cc4ffae630ff

mapk committed:
update-core.php
Minor design changes including WP logo size and padding/margins of type columns

aaronjorbin commented on tests/qunit/index.html:
```diff
@@ -59,6 +61,7 @@
 		<script src="wp-admin/js/customize-nav-menus.js"></script>
 		<script src="wp-admin/js/customize-widgets.js"></script>
 		<script src="wp-admin/js/word-count.js"></script>
+		<script src="wp-admin/js/tests.js"></script>
```
This name should match the file in wp-admin that is being tested. I believe `updates.js` is what is being tested here.

swissspidy committed:
Fix file name used in QUnit tests

Rename forgotten in 0502ad43e48d0d240f93f41752c48aa458ee885a

swissspidy committed:
Merge pull request #8 from mapk/shiny-updates

Adds minor design changes to update-core.php

obenland committed:
Error handler updates

* Method documentation.
* More descriptive variables.
* Handle updating of inline update messages.

swissspidy committed:
Only add items to queue if not already in it

See obenland/shiny-updates#268

obenland committed:
Text > html

obenland committed:
Give immediate feedback after an update button was clicked.

Also prevents multiple updates of the same item on multiple button
clicks.

Reverts 2ff5f7f629eafcb4e6467acdca0e1eabf49f5789.

obenland committed:
Remove update-core bits.

They were not approved for merge.

obenland committed:
Revert IDE changes.

obenland committed:
Revert IDE changes.

swissspidy committed:
Merge pull request #9 from swissspidy/remove-update-core

Remove update-core bits.

ocean90 commented on src/wp-admin/css/list-tables.css:
```diff
@@ -1678,6 +1637,14 @@ div.action-links,
+
+.wordpress-reinstall-card h2 {
+	margin-top: 1em;
+}
```
This should be removed too.

ocean90 commented on src/wp-admin/includes/plugin-install.php:
```diff
@@ -435,6 +435,8 @@ function install_plugin_install_status($api, $loop = false) {
 	global $tab;
 
+	remove_action( 'install_plugins_pre_plugin-information', 'install_plugin_information' );
+
```
Is this still needed?

ocean90 commented on src/wp-admin/includes/update.php:
```diff
@@ -309,65 +309,69 @@ function wp_plugin_update_rows() {
-			/* translators: 1: plugin name, 2: details URL, 3: accessibility text, 4: version number */
-			printf( __( 'There is a new version of %1$s available. <a href="%2$s" class="thickbox open-plugin-details-modal" aria-label="%3$s">View version %4$s details</a>.' ),
+			/* translators: 1: plugin name, 2: details URL, 3: escaped plugin name, 4: version number */
+			printf( __( 'There is a new version of %1$s available. <a href="%2$s" class="thickbox open-plugin-details-modal" aria-label="View %3$s version %4$s details">View version %4$s details</a>.' ),
```
Was this a merge conflict or why is the accessibility text moved back to the string?

obenland committed:
Final bits of update-core cleanup.

obenland committed:
Remove merge conflicts with changes introduced in https://core.trac.wordpress.org/changeset/36844.

Also adds inline docs for `$wp_list_table`.

obenland committed:
Bring index and modal update messages N'Sync.

Fixes obenland/shiny-updates#280.

obenland committed:
Nothing more todo here.

swissspidy committed:
Merge branch 'shiny-updates' into generic-error-handler

obenland committed:
Prevent search submit from actually submitting the form.

obenland committed:
Escape search values.

obenland committed:
Update URL with search param and make it work with typeselector.

swissspidy committed:
Change the error message depending on the action

It will read "Update failed: xy" or "Installation failed: xy" now

obenland committed:
Tabs > spaces.

swissspidy committed:
errorReason -> error

swissspidy committed:
Add some whitespace

swissspidy committed:
Fix fixtures

obenland committed:
Merge pull request #5 from swissspidy/generic-error-handler

Basic validation of AJAX responses.

Fixes obenland/shiny-updates#272.

obenland committed:
Use WordPress' standard debounce time

obenland committed:
Search improvements.

Only start a new search request if the term changed. Also adds all now
necessary query args to the URL on plugin-install.

obenland committed:
Restore update-core CSS selector

obenland committed:
Merge remote-tracking branch 'aaronjorbin/master' into shiny-updates

swissspidy committed:
Revert "Revert c2ca7fc9d15686759b6c0ac1fd24cc4ffae630ff"

This reverts commit af4bee4278056b3e04432802e12f0d0e13c52403.

obenland committed:
joshing fixes

obenland committed:
Use appropriate description

swissspidy committed:
Pass `pagenow` explicitly instead of relying on the implicit global declaration

aaronjorbin commented on tests/qunit/fixtures/updates.js:
```diff
@@ -0,0 +1,53 @@
+	'ajax': {
+		'url': '\/wp-admin\/admin-ajax.php'
+	}
+};
```
You can add the global `pagenow` here.

obenland replied:
Adding it there, does it fix the errors for you? What did you set it to?

obenland committed:
Revert "Pass `pagenow` explicitly instead of relying on the implicit global declaration"

This reverts commit ec426fdf3f54c8f779669636ed9b4af008b43c66.
